source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -421,6 +425,59 @@ void SetupContextForAmazonServiceAccount(string @namespace, string clusterUrl, s
                 var clusterName = variables.Get(SpecialVariables.EksClusterName);
                 log.Info($"Creating kubectl context to {clusterUrl} (namespace {@namespace}) using EKS cluster name {clusterName}");
 
+                if (!TrySetAws())
+                {
+                    log.Warn("Could not find the aws cli, falling back to the aws-iam-authenticator.");
+                }
+                else
+                {
+                    try
+                    {
+                        var region = Regex.Match(clusterUrl, @"(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d", RegexOptions.IgnoreCase).Value;
+                        if (string.IsNullOrWhiteSpace(region))
+                        {
+                            log.Error("The EKS cluster Url specified should contain a valid aws region name");
+                        }
+                        
+                        var awsCliCommandRes = ExecuteCommandAndReturnOutput(aws, "--version").FirstOrDefault();
+                        var awsCliVersion = Regex.Match(awsCliCommandRes, @"aws-cli\/(\d+\.)?(\d+\.)?(\*|\d+)").Value.Replace("aws-cli/", string.Empty);
```

Comment:
Here we are not using different parts of SemVer, but just treat it as a whole string. Therefore it is not necessary to use regex here since it is a bit less easy to read. The `aws --version` command prints the versions of dependencies on a space separated line, eg:
```
aws-cli/2.8.4 Python/3.9.11 Darwin/21.4.0 exe/x86_64 prompt/off
```
We can simply split it:
```cs
var awsCliVersion = awsCliCommandRes.Split()
    .FirstOrDefault(versions => versions.StartsWith("aws-cli"))
    .Replace("aws-cli/", string.Empty);
```